projects/storefrontlib/src/layout/launch-dialog/services/launch-dialog.service.ts

```diff
@@ -4,17 +4,20 @@ import {
   Inject,
   Injectable,
   isDevMode,
+  OnDestroy,
   ViewContainerRef,
 } from '@angular/core';
 import { resolveApplicable } from '@spartacus/core';
-import { BehaviorSubject, combineLatest, Observable } from 'rxjs';
+import { BehaviorSubject, combineLatest, Observable, Subscription } from 'rxjs';
+import { filter, map, take, tap } from 'rxjs/operators';
 import { LayoutConfig } from '../../config/layout-config';
 import { LaunchOptions, LAUNCH_CALLER } from '../config/launch-config';
 import { LaunchRenderStrategy } from './launch-render.strategy';
-import { filter, map, tap } from 'rxjs/operators';
 
 @Injectable({ providedIn: 'root' })
-export class LaunchDialogService {
+export class LaunchDialogService implements OnDestroy {
+  protected subscription = new Subscription();
```

Comment:
Interface `OnDestroy` is only for Components/Directives, or Services which are [direct provider](https://coryrylan.com/blog/using-ngondestroy-with-services-in-angular), method `ngOnDestroy` will never call here.
Anyway, in the case when we use the `take` operator, unsubscribe is optional/redundant. So I propose to don't care about it in this case

Reply:
Agreed.